docs/components/LinkComponent.tsx

```diff
@@ -14,10 +14,13 @@ export const LinkComponent = forwardRef<HTMLAnchorElement, LinkComponentProps>(
 		// Use an `a` tag when linking externally
 		// Regex finds links starting with: `http://` | `https://` | `//`
 		const hrefAsString = typeof href === 'string' ? href : href?.pathname;
-		if (hrefAsString && /^(https?:\/\/|\/\/)/i.test(hrefAsString)) {
+		if (
+			(hrefAsString && /^(https?:\/\/|\/\/)/i.test(hrefAsString)) ||
+			hrefAsString?.includes('#')
```

Comment:
Next's Link doesn't pass focus when using in page links. This change allows the in page linking to work.